Source/SwiftLintBuiltInRules/Rules/Idiomatic/OneDelarationPerFileRule.swift

```diff
@@ -0,0 +1,118 @@
+        description: "One declaration per file is allowed, extensions are an exception",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    """),
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    extension Car {
+                        func drive() {
+                        }
+                        func stop() {
+                        }
+                    }
+                    """)
+        ],
+        triggeringExamples: [
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    class Bike {
+                        func ride() {
+                        }
+                        func stop() {
+                        }
+                    }
+                    """),
+            Example("""
+                    protocol Identifiable {
+                        var identifier: String { get }
+                    }
+                    enum IdentifiableTypes: String {
+                        case linear, composite
+                    }
+                    """),
+            Example("""
+                    struct BasicProfile {
+                        var id: Int
+                        var name: String
+                    }
+                    struct DetailedProfile {
+                        var basic: BasicProfile
+                        var age: String
+                        var genderRaw: String
+                    }
+                    """)
+        ]
+    )
+}
+
+private extension OneDelarationPerFileRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        var declarationsCount: Int = 0
+        override var skippableDeclarations: [any DeclSyntaxProtocol.Type] {
+            return .allExcept(ClassDeclSyntax.self, StructDeclSyntax.self, EnumDeclSyntax.self, ProtocolDeclSyntax.self)
+        }
+
+        override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+            declarationsCount += 1
+            appendViolationIfNeeded(node: node.name)
```

Comment:
The rule should preferably trigger on the declaration-introducing keywords. After all, it's the declarations that violate, not the names.
```suggestion
            appendViolationIfNeeded(node: classKeyword)
```